File: app/src/api/product-catalog/data-transfer-objects/review.ts
import { PadCut } from './pad-cut';
import { PadFinish } from './pad-finish';
import { Stars } from './stars';

export class Review {
    constructor(
        public username: string,
        public date: Date,
        public stars: Stars,
        public cut: PadCut | null,
        public finish: PadFinish | null,
        public title: string,
        public body: string
    ) { }
}

File: api/src/Domain/ProductCatalog/Entities/Review.cs
using System;
using DetailingArsenal.Domain.Users;

namespace DetailingArsenal.Domain.ProductCatalog {
    public class Review : Aggregate<Review>, IUserEntity {
        public const int TitleMaxLength = 64;
        public const int BodyMaxLength = 10_000;

        public Guid UserId { get; }
        public Guid PadId { get; }
        public DateTime CreatedDate { get; }
        public int Stars { get; set; }
        public int? Cut { get; set; }
        public int? Finish { get; set; }
        public string Title { get; set; }
        public string Body { get; set; }

        public Review(Guid userId, Guid padId, DateTime createdDate, int stars, int? cut, int? finish, string title, string body) {
            Id = Guid.NewGuid();
            UserId = userId;
            PadId = padId;
            CreatedDate = createdDate;
            Stars = stars;
            Cut = cut;
            Finish = finish;
            Title = title;
            Body = body;
        }
    }
}